docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/Ranking/LightGbmWithOptions.cs

```diff
@@ -1,51 +1,121 @@
         {
-            // Creating the ML.Net IHostEnvironment object, needed for the pipeline.
-            var mlContext = new MLContext();
+            // Create a new context for ML.NET operations. It can be used for exception tracking and logging, 
+            // as a catalog of available operations and as the source of randomness.
+            // Setting the seed to a fixed number in this example to make outputs deterministic.
+            var mlContext = new MLContext(seed: 0);
 
-            // Download and featurize the train and validation datasets.
-            var dataview = Microsoft.ML.SamplesUtils.DatasetUtils.LoadFeaturizedMslrWeb10kDataset(mlContext);
+            // Create a list of training data points.
+            var dataPoints = GenerateRandomDataPoints(1000);
 
-            // Leave out 10% of the dataset for testing. Since this is a ranking problem, we must ensure that the split
-            // respects the GroupId column, i.e. rows with the same GroupId are either all in the train split or all in
-            // the test split. The samplingKeyColumn parameter in Data.TrainTestSplit is used for this purpose.
-            var split = mlContext.Data.TrainTestSplit(dataview, testFraction: 0.1, samplingKeyColumnName: "GroupId");
+            // Convert the list of data points to an IDataView object, which is consumable by ML.NET API.
+            var trainingData = mlContext.Data.LoadFromEnumerable(dataPoints);
 
-            // Create the Estimator pipeline. For simplicity, we will train a small tree with 4 leaves and 2 boosting iterations.
-            var pipeline = mlContext.Ranking.Trainers.LightGbm(
-                new LightGbmRankingTrainer.Options
+            // Define trainer options.
+            var options = new LightGbmRankingTrainer.Options
+            {
+                NumberOfLeaves = 4,
+                MinimumExampleCountPerGroup = 10,
+                LearningRate = 0.1,
+                NumberOfIterations = 2,
+                Booster = new GradientBooster.Options
                 {
-                    NumberOfLeaves = 4,
-                    MinimumExampleCountPerGroup = 10,
-                    LearningRate = 0.1,
-                    NumberOfIterations = 2,
-                    Booster = new GradientBooster.Options
-                    {
-                        FeatureFraction = 0.9
-                    },
-                    RowGroupColumnName = "GroupId"
-                });
-
-            // Fit this pipeline to the training Data.
-            var model = pipeline.Fit(split.TrainSet);
-
-            // Evaluate how the model is doing on the test data.
-            var dataWithPredictions = model.Transform(split.TestSet);
-
-            var metrics = mlContext.Ranking.Evaluate(dataWithPredictions);
-            Microsoft.ML.SamplesUtils.ConsoleUtils.PrintMetrics(metrics);
+                    FeatureFraction = 0.9
+                },
+                RowGroupColumnName = "GroupId"
```

Comment:
>RowGroupColumnName = "GroupId" [](start = 16, length = 30)

I find it strange what we use this line only in one example out of 4.
If it's important, why not in all 4, if it's not, why we even show it?

Reply:
So in the two samples without option, the `RowGrouColumnName` column defaults to 'GroupID'.
However, I just noticed that in the advanced options constructor, the `RowGroupColumnName` defaults to null. We should probably fix this.

---
In reply to: [275931195](https://github.com/dotnet/machinelearning/pull/3338#discussion_r275931195) [](ancestors = 275931195)

Reply:
In the meantime I am going to include this line in the other sample with options.

---
In reply to: [275985563](https://github.com/dotnet/machinelearning/pull/3338#discussion_r275985563) [](ancestors = 275985563,275931195)